### ISSUE TITLE
Creating replica in hierarchy with leaf coordinating resource results in unhelpful errors

### ISSUE THREAD

**Issue by alanking** [irods/irods#5398]
### iRODS Version, OS and Version
at least 4.2.8, probably all versions

### What did you try to do?
Put a file to a resource hierarchy which has no storage resource at the leaf.

### Expected behavior
A message saying that the coordinating resource has missing children, *which* coordinating resource failed to find its children, and an appropriate error code (`CHILD_NOT_FOUND`?)
```
$ iput -R cliff irodsctl goo
remote addresses: 172.17.0.2 ERROR: putUtil: put error for /tempZone/home/rods/goo, status = -1804000 status = -1804000 CHILD_NOT_FOUND
Level 0: Operation not permitted - resource [dangle] is a coordinating resource with no children. Full hierarchy: [cliff;dangle]
```

### Observed behavior (including steps to reproduce, if applicable)
The result depends on the type of resource and the operation. When the hierarchy ends in a passthru:
```
$ ilsresc | head -n2
cliff:passthru
└── dangle:passthru
$ iput -R cliff irodsctl foo
remote addresses: 172.17.0.4 ERROR: putUtil: put error for /tempZone/home/rods/foo, status = -1 status = -1 Unknown iRODS error, Operation not permitted
```
When the hierarchy ends in a replication resource, you get something a little closer to the ideal:
```
$ iput -R cliff irodsctl goo
remote addresses: 172.17.0.2 ERROR: putUtil: put error for /tempZone/home/rods/goo, status = -1804000 status = -1804000 CHILD_NOT_FOUND
```

Basically, we just need things to be consistent.
